### PR TITLE
WFLY-11479_HTTP_Basic_auth_silent_mode

### PR DESCRIPTION
**Pull request by Skyllarr**

### CODE REVIEW

File: security/WFLY-11479_HTTP_Basic_auth_silent_mode.adoc [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+== Overview
+
+Using HTTP Basic authentication with Undertow it was possible to set silent mode. Basic authentication in silent mode will send a challenge only if the request contained authorization header, otherwise it is assumed another method will send the challenge. This behaviour will allow to combine basic auth with form auth, so human users will use form based auth and programmatic clients can use basic authentication to log in. This RFE is to support the same behaviour in Elytron.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-11479[WFLY-11479]
```
Comment (mchoma) [inline security/WFLY-11479_HTTP_Basic_auth_silent_mode.adoc]: I think there was also ELY issue with forum post reference

File: security/WFLY-11479_HTTP_Basic_auth_silent_mode.adoc [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1154[EAP7-1154]
+
+=== Dev Contacts
+
+* mailto:dvilkola@redhat.com[dvilkola]
+
+=== QE Contacts
```
Comment (mchoma) [inline security/WFLY-11479_HTTP_Basic_auth_silent_mode.adoc]: you can fill in me

File: security/WFLY-11479_HTTP_Basic_auth_silent_mode.adoc [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+=== Other Interested Projects
+
+Undertow Core - Basic authentication in Undertow has the same mode available
+
+== Requirements
+
+=== Hard Requirements
+
+This will involve processing optional property `silent` set along with basic authentication. If the property is set as true and request does not contain Authorization header, response will not include `WWW-Authenticate` header and http status code will not be `401`.
```
Comment (mchoma) [inline security/WFLY-11479_HTTP_Basic_auth_silent_mode.adoc]: Where will be this property? I suppose mechanism configuration property in generic configuration map? So this RFE does no involve subsystem change?

File: security/WFLY-11479_HTTP_Basic_auth_silent_mode.adoc [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+== Test Plan
```
Comment (mchoma) [inline security/WFLY-11479_HTTP_Basic_auth_silent_mode.adoc]: Do you plan to add some tests?

File: security/WFLY-11479_HTTP_Basic_auth_silent_mode.adoc (hunk 3 above)
Comment (Skyllarr) [reply security/WFLY-11479_HTTP_Basic_auth_silent_mode.adoc]: Yes, in generic configuration map. It involves changing only BasicAuthenticationMechanism.java class. Implementation is here: https://github.com/Skyllarr/wildfly-elytron/commit/e221c79a234fde92bdb63bb041a922c0b077a0ad (was thinking I will create PR after adding tests if necessary)

File: security/WFLY-11479_HTTP_Basic_auth_silent_mode.adoc (hunk 4 above)
Comment (Skyllarr) [reply security/WFLY-11479_HTTP_Basic_auth_silent_mode.adoc]: if tests are necessary, I will add them. The code is very similar to implementation in Undertow here: https://github.com/undertow-io/undertow/blob/master/core/src/main/java/io/undertow/security/impl/BasicAuthenticationMechanism.java

Comment (darranl) [reply security/WFLY-11479_HTTP_Basic_auth_silent_mode.adoc]: We do have some HTTP auth tests with Elytron, I would expect this should be simple to add there.

File: security/WFLY-11479_HTTP_Basic_auth_silent_mode.adoc [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+:email:             dvilkola@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          basic,elytron,silent
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Using HTTP Basic authentication with Undertow it was possible to set silent mode. Basic authentication in silent mode will send a challenge only if the request contained authorization header, otherwise it is assumed another method will send the challenge. This behaviour will allow to combine basic auth with form auth, so human users will use form based auth and programmatic clients can use basic authentication to log in. This RFE is to support the same behaviour in Elytron.
```
[the remark this answers is not in the document]
Comment (darranl) [reply security/WFLY-11479_HTTP_Basic_auth_silent_mode.adoc]: I would expect that to work.

File: security/WFLY-11479_HTTP_Basic_auth_silent_mode.adoc (hunk 4 above)
Comment (Skyllarr) [reply security/WFLY-11479_HTTP_Basic_auth_silent_mode.adoc]: @darranl thank you, I'll add tests and will update analysis doc

**Pull request merged**